server/neptune/workflows/internal/deploy/revision/queue/pull_rebaser.go

```diff
@@ -0,0 +1,159 @@
+	// TODO: Add alarms on GithubListOpenPRs and GithubListModifiedFiles API call failures
+	ghOpts := originalOpts
+	ghOpts.ScheduleToCloseTimeout = RebaseGithubScheduleToCloseTimeout
+	ctx = workflow.WithActivityOptions(ctx, ghOpts)
+
+	// list open PRs
+	var listOpenPRsResp activities.ListOpenPRsResponse
+	err := workflow.ExecuteActivity(ctx, p.GithubActivities.GithubListOpenPRs, activities.ListOpenPRsRequest{
+		Repo: repo,
+	}).Get(ctx, &listOpenPRsResp)
+	if err != nil {
+		return errors.Wrap(err, "listing open PRs")
+	}
+
+	// spawn activities to list modified fles in each open PR in parallel
+	futureByPullNum := map[github.PullRequest]workflow.Future{}
+	for _, pullRequest := range listOpenPRsResp.PullRequests {
+		future := workflow.ExecuteActivity(ctx, p.GithubActivities.GithubListModifiedFiles, activities.ListModifiedFilesRequest{
+			Repo:        repo,
+			PullRequest: pullRequest,
+		})
+		futureByPullNum[pullRequest] = future
+	}
+
+	// resolve the futures and check if an open PR needs to be rebased
+	prsToRebase := []github.PullRequest{}
+	for pullRequest, future := range futureByPullNum {
+		var result activities.ListModifiedFilesResponse
+
+		// list modified files should not fail unless we hit the ratelimit which should autoresolve once our ratelimit revives in an hour max
+		// If it errors out due to any other reason, let's be preventive and rebase this PR as well
+		listFilesErr := future.Get(ctx, &result)
+		if listFilesErr != nil {
+			logger.Error(ctx, "error listing modified files in PR", key.ErrKey, listFilesErr, "pull_num", pullRequest.Number)
+			prsToRebase = append(prsToRebase, pullRequest)
+			continue
+		}
+
+		shouldRebase, err := shouldRebasePullRequest(root, result.FilePaths)
+
+		// unlikey for error since we validate the WhenModified config at startup
+		// if it errors out, let's be preventive and rebase this PR as well
+		if err != nil {
+			logger.Error(ctx, "error matching filepaths in PR", key.ErrKey, err, "pull_num", pullRequest.Number)
+			prsToRebase = append(prsToRebase, pullRequest)
+			continue
+		}
+
+		if shouldRebase {
+			prsToRebase = append(prsToRebase, pullRequest)
+		}
+	}
+
+	// skip set pr revision call if no prs to rebase
+	if len(prsToRebase) == 0 {
+		return nil
+	}
+
+	// reset workflow options
+	ctx = workflow.WithActivityOptions(ctx, originalOpts)
+	futureByPullNum = map[github.PullRequest]workflow.Future{}
+	for _, pr := range prsToRebase {
+		futureByPullNum[pr] = workflow.ExecuteActivity(ctx, p.PrRevisionSetterActivities.SetPRRevision, activities.SetPRRevisionRequest{
+			Repository:  repo,
+			PullRequest: pr,
+		})
+	}
+
+	// TODO: Add alarms on SetPRRevision activity failures
+	for pr, future := range futureByPullNum {
+		var resp activities.SetPRRevisionResponse
+		err := future.Get(ctx, &resp)
+		if err != nil {
+			logger.Error(ctx, "error making api call to set pr revision", key.ErrKey, err, "pull_num", pr.Number)
```

Comment:
We create keys for everything we use, can you do that for pull instead of just a raw string?

Reply:
Also we should return the error, the caller should determine what they want to do with it.